src/transformers/models/qwen2_vl/modeling_qwen2_vl.py

```diff
@@ -1860,3 +1861,173 @@ def prepare_inputs_for_generation(
             }
         )
         return model_inputs
+
+
+@add_start_docstrings(
+    """(Inspired by the LlamaForSequenceClassification)
+    The Qwen2-VL transformer with a sequence classification head on top (linear layer).
+    Copied the Qwen2VLForConditionalGenerations forward method, but with modification such that it supports sequence classification using the last token embedding instead.
+
+    [`Qwen2VLForSequenceClassification`] uses the last token in order to do the classification, as other causal models
+    (e.g. GPT-2) do.
+
+    Since it does classification on the last token, it requires to know the position of the last token. If a
+    `pad_token_id` is defined in the configuration, it finds the last token that is not a padding token in each row. If
+    no `pad_token_id` is defined, it simply takes the last value in each row of the batch. Since it cannot guess the
+    padding tokens when `inputs_embeds` are passed instead of `input_ids`, it does the same (take the last value in
+    each row of the batch).
+    """,
+    QWEN2VL_START_DOCSTRING,
+)
+class Qwen2VLForSequenceClassification(Qwen2VLPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.visual = Qwen2VisionTransformerPretrainedModel._from_config(
+            config.vision_config, attn_implementation=config._attn_implementation
+        )
+        self.model = Qwen2VLModel(config)
+        self.vocab_size = config.vocab_size
+        self.classification_head = nn.Linear(config.hidden_size, self.num_labels, bias=False)
+        self.padding_side = "left"  # set it to left by default, user can use setter to change padding_sides
```

Comment:
we usually try to not set padding side within modeling code. I realize this is copied from `Qwen2VLConditionalGeneration` but since it is not used anywhere, let's remove it

Reply:
Alright. If the model was trained like this, won't it hurt performance to change the padding side to right side though?

Reply:
Alright, thanks for clarifying!